catroid/src/androidTest/java/org/catrobat/catroid/test/content/actions/GoNStepsBackActionTest.java

```diff
@@ -142,52 +146,59 @@ public void testBoundarySteps() {
 		parentGroup.addActor(background.look);
 		assertEquals(0, background.look.getZIndex());
 
-		Sprite sprite = new SingleSprite("testSprite");
+		Sprite sprite = new SingleSprite("PenActor");
 		parentGroup.addActor(sprite.look);
 		assertEquals(1, sprite.look.getZIndex());
 
-		Sprite sprite2 = new SingleSprite("testSprite2");
+		Sprite sprite2 = new SingleSprite("EmbroideryActor");
 		parentGroup.addActor(sprite2.look);
 		assertEquals(2, sprite2.look.getZIndex());
 
+		Sprite sprite3 = new SingleSprite("testSprite");
+		parentGroup.addActor(sprite3.look);
+		assertEquals(Constants.Z_INDEX_FIRST_SPRITE, sprite3.look.getZIndex());
+
 		project.getDefaultScene().addSprite(sprite);
 		project.getDefaultScene().addSprite(sprite2);
 		ProjectManager.getInstance().setProject(project);
 
 		sprite.getActionFactory().createGoNStepsBackAction(sprite, new Formula(Integer.MAX_VALUE)).act(1.0f);
-		assertEquals(Constants.Z_INDEX_FIRST_SPRITE, sprite.look.getZIndex());
 		assertEquals(1, sprite2.look.getZIndex());
 
-		sprite.getActionFactory().createGoNStepsBackAction(sprite, new Formula(Integer.MIN_VALUE)).act(1.0f);
-		assertEquals(2, sprite.look.getZIndex());
+		sprite.getActionFactory().createGoNStepsBackAction(sprite2, new Formula(Integer.MIN_VALUE)).act(1.0f);
+		assertEquals(3, sprite2.look.getZIndex());
 	}
 
 	@Test
 	public void testBrickWithStringFormula() {
-		sprite.getActionFactory().createGoNStepsBackAction(sprite2, new Formula(String.valueOf(STEPS))).act(1.0f);
+		sprite.getActionFactory().createGoNStepsBackAction(sprite3, new Formula(String.valueOf(STEPS))).act(1.0f);
 		assertEquals(0, background.look.getZIndex());
```

Comment:
please use variables for the 0,1,2,etc.

also, think about if you really want to use the Constants.Z_INDEX_FIRST_SPRITE from constants, or if you want to have it defined in your test. what kind of bugs / misbehavior do you want to find with this test? if someone changes Constants.Z_INDEX_FIRST_SPRITE to 0 or 27 in constants, this test would probably still work.
also it would be a possibility to add a test for the layer order, to verify that the different sprites ZIndex is in the correct order. ("real" sprites always in the topmost foreground)